.pre-commit-config.yaml

```diff
@@ -0,0 +1,56 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-added-large-files
+      - id: check-merge-conflict
+  # Formatters: hooks that re-write files
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: v1.3.5
+    hooks:
+      - id: clang-format
+        args: ["-i", "--style=file"]
```

Comment:
The pre-commit clang-format disagrees with the fix-style clang-format. Probably because you moved the `.clang-format` configuration around but also there is some actual reformatting taking place in the cpp files. Can you please verify? It would be good to reformatting to a minimum.

Reply:
I also wonder where the discrepancy comes from. I do not think that this is due to the new location of `.clang-format` because its rules seem to be applied still. Do we like to keep the `fix-style` workflow? If not, this discrepancy might not be a big issue as it anyway just affects two cpp files in the `pairinteraction` folder. The other files are not touched, so reformatting is not happening too often.

Reply:
We can get rid of fix-style if pre-commit can take its place, i.e. fail on insufficient formatting.

Reply:
Ok, then I will remove fix-style, as pre-commit fails on insufficient formatting.

Reply:
Oh, and it needs to generate a patch that one can copy and apply  in case of failure:
https://github.com/pairinteraction/pairinteraction/blob/5873e99063d6d053d7d919b4ada9a86a9a86aea4/.github/workflows/linux.yml#L145-L150